Index: 3.6.1-8c1.patch
===================================================================
--- 3.6.1-8c1.patch	(révision 114)
+++ 3.6.1-8c1.patch	(copie de travail)
@@ -1,104 +0,0 @@
-Index: gcode.cpp
-===================================================================
---- gcode.cpp	(révision 120)
-+++ gcode.cpp	(copie de travail)
-@@ -849,6 +849,18 @@
-     return status;
- }
- 
-+/*
-+/// LETARTARE  May 13, 2014
-+	1- frame1 : < 8c (3 axes), 0.845 (4 axes)  	-> $$==0
-+		received == "MPos:[....],WPos:[....]"
-+	2- frame2 : 0.8c1 (4 axes)       			-> $$==1
-+		received == "<State,MPos:...,WPos:...>"
-+    3- frame2 : 0.8c (3 axes)        			-> $$==1
-+		received == "<State,MPos:...,WPos:...>"
-+    4- frame3 : 0.9d (3 axes), 0.9d1 (4 axes) 	-> $$==1
-+		received == "<State,MPos: ... ,WPos:...,Ln=..>"
-+*/
-+
- void GCode::parseCoordinates(const QString& received, bool aggressive)
- {
-     if (aggressive)
-@@ -860,7 +872,7 @@
-         parseCoordTimer.restart();
-     }
- 
--	bool good = false;
-+	bool good = false, goodMPos = false;
- 	int captureCount ;
- 	QString state;
-     QString prepend;
-@@ -875,25 +887,38 @@
- 	QString coordRegExp;
- 	QRegExp rxStateMPos;
- 	QRegExp rxWPos;
--	/// 3 axis
--	QString format("(-*\\d+\\.\\d+),(-*\\d+\\.\\d+)") ;
--    int maxaxis = MAX_AXIS_COUNT, naxis ;
--    for (naxis = DEFAULT_AXIS_COUNT; naxis <= maxaxis; naxis++) {
-+/// 1 axis
-+	QString format1("(-*\\d+\\.\\d+)") ;
-+	QString sep(",");
-+	/// 3 axes
-+	QString format3 = format1 + sep + format1 + sep + format1 ;
-+	/// 4 axes
-+	QString format4 = format3 + sep + format1 ;
-+	QString format ;
-+    int naxis ;
-+	for (naxis = MAX_AXIS_COUNT; naxis >= DEFAULT_AXIS_COUNT; naxis--) {
- 		if (!doubleDollarFormat)
- 			captureCount = naxis ;
- 		else
- 			captureCount = naxis + 1 ;
--		//
--		format += ",(-*\\d+\\.\\d+)" ;
-+
-+               if (naxis == MAX_AXIS_COUNT )
-+			format = format4;
-+		else
-+			format = format3;
-+
- 		coordRegExp = prepend + format + append ;
- 		rxStateMPos = QRegExp(preamble + coordRegExp);
--		rxWPos = QRegExp(QString("WPos:") + coordRegExp);
--		good = rxStateMPos.indexIn(received, 0) != -1
--			   && rxStateMPos.captureCount() == captureCount
--			   && rxWPos.indexIn(received, 0) != -1
--			   && rxWPos.captureCount() == naxis
--			   ;
--		// find ...
-+                goodMPos = rxStateMPos.indexIn(received, 0) != -1
-+					&& rxStateMPos.captureCount() == captureCount
-+					;
-+		if (goodMPos) {
-+			rxWPos = QRegExp(QString("WPos:") + coordRegExp);
-+			good = rxWPos.indexIn(received, 0) != -1
-+					&& rxWPos.captureCount() == naxis
-+					;
-+}
-+		// find naxis ...
- 		if (good)
- 			break;
- 	}
-@@ -941,15 +966,15 @@
- 			workCoord.stoppedZ = false;
- 
- 		workCoord.sliderZIndex = sliderZCount;
-+if (doubleDollarFormat)
-+			diag(qPrintable(tr("Decoded: State:%s")),  qPrintable(state) );
-         if (numaxis == DEFAULT_AXIS_COUNT)
--			diag(qPrintable(tr("Decoded: State:%s MPos: %f,%f,%f WPos: %f,%f,%f\n")),
--				 qPrintable(state),
-+diag(qPrintable(tr("Decoded: MPos: %f,%f,%f WPos: %f,%f,%f\n")),			
- 				 machineCoord.x, machineCoord.y, machineCoord.z,
- 				 workCoord.x, workCoord.y, workCoord.z
- 				 );
-         else if (numaxis == MAX_AXIS_COUNT)
--			diag(qPrintable(tr("Decoded: State:%s MPos: %f,%f,%f,%f WPos: %f,%f,%f,%f\n")),
--				 qPrintable(state),
-+diag(qPrintable(tr("Decoded: MPos: %f,%f,%f,%f WPos: %f,%f,%f,%f\n")),
-                  machineCoord.x, machineCoord.y, machineCoord.z, machineCoord.fourth,
-                  workCoord.x, workCoord.y, workCoord.z, workCoord.fourth
- 				 );
Index: gcode.cpp
===================================================================
--- gcode.cpp	(révision 114)
+++ gcode.cpp	(copie de travail)
@@ -163,6 +163,8 @@
             // it is possible that we are already connected and missed the
             // signon banner. Force a reset (is this ok?) to get the banner
 
+/// LETARTARE for 0.845 !!
+/*
             emit addListOut("(CTRL-X)");
 
             char buf[2] = {0};
@@ -178,7 +180,8 @@
                 emit sendMsg(msg);
                 return;
             }
-
+*/
+/// <--
             if (!waitForStartupBanner(result, SHORT_WAIT_SEC, true))
                 return;
         }
@@ -307,7 +310,7 @@
 {
     if (result.contains("Grbl"))
     {
-        QRegExp rx("Grbl (\\d+)\\.(\\d+)(\\w*)");
+      QRegExp rx("Grbl (\\d+)\\.(\\d+)(\\w*)(\\d*)");
         if (rx.indexIn(result) != -1 && rx.captureCount() > 0)
         {
             doubleDollarFormat = false;
@@ -318,7 +321,7 @@
                 int majorVer = list.at(1).toInt();
                 int minorVer = list.at(2).toInt();
                 char letter = ' ';
-                if (list.size() == 4 && list.at(3).size() > 0)
+                if (list.size() >= 4 && list.at(3).size() > 0)
                 {
                     letter = list.at(3).toLatin1().at(0);
                 }
@@ -330,6 +333,10 @@
 
                 diag(qPrintable(tr("Got Grbl Version (Parsed:) %d.%d%c ($$=%d)\n")),
                             majorVer, minorVer, letter, doubleDollarFormat);
+/// LETARTARE
+                QString resu = list.at(0);
+				emit setVersionGrbl(resu);
+/// <--
             }
 
             if (!doubleDollarFormat)
@@ -384,6 +391,10 @@
     }
     else if (!sentReqForLocation)// if requesting location, don't add that "noise" to the output view
     {
+/// LETARTARE
+    	if (currLine && (line.at(0).toLatin1() != 'N') )
+			line = "L" + QString().setNum(currLine) + "  " + line;
+/// <--
         emit addListOut(line);
     }
 
@@ -872,7 +883,7 @@
         parseCoordTimer.restart();
     }
 
-	bool good = false, goodMPos = false;
+	bool good = false ;
 	int captureCount ;
 	QString state;
     QString prepend;
@@ -909,15 +920,13 @@
 
 		coordRegExp = prepend + format + append ;
 		rxStateMPos = QRegExp(preamble + coordRegExp);
-                goodMPos = rxStateMPos.indexIn(received, 0) != -1
-					&& rxStateMPos.captureCount() == captureCount
-					;
-		if (goodMPos) {
-			rxWPos = QRegExp(QString("WPos:") + coordRegExp);
-			good = rxWPos.indexIn(received, 0) != -1
-					&& rxWPos.captureCount() == naxis
-					;
-}
+ rxWPos = QRegExp(QString("WPos:") + coordRegExp);
+		good = rxStateMPos.indexIn(received, 0) != -1
+				&& rxStateMPos.captureCount() == captureCount
+				&& rxWPos.indexIn(received, 0) != -1
+				&& rxWPos.captureCount() == naxis
+				;
+
 		// find naxis ...
 		if (good)
 			break;
@@ -966,6 +975,7 @@
 			workCoord.stoppedZ = false;
 
 		workCoord.sliderZIndex = sliderZCount;
+
 if (doubleDollarFormat)
 			diag(qPrintable(tr("Decoded: State:%s")),  qPrintable(state) );
         if (numaxis == DEFAULT_AXIS_COUNT)
@@ -1321,6 +1331,9 @@
         }
         else if (s.at(0) == 'X' || s.at(0) == 'Y' || s.at(0) == 'Z'
                  || s.at(0) == 'A' || s.at(0) == 'B' || s.at(0) == 'C'
+/// LETARTARE
+				|| s.at(0) == 'U' || s.at(0) == 'V' || s.at(0) == 'W'
+/// <--
                  || s.at(0) == 'I' || s.at(0) == 'J' || s.at(0) == 'K'
                  || s.at(0) == 'F' || s.at(0) == 'L' || s.at(0) == 'S')
         {
Index: mainwindow.cpp
===================================================================
--- mainwindow.cpp	(révision 114)
+++ mainwindow.cpp	(copie de travail)
@@ -44,7 +44,7 @@
 
     readSettings();
 
-    info("%s has started", GRBL_CONTROLLER_NAME_AND_VERSION);
+    info(qPrintable(tr("%s has started")), GRBL_CONTROLLER_NAME_AND_VERSION);
 
     // see http://blog.qt.digia.com/2010/06/17/youre-doing-it-wrong/
     // The thread points out that the documentation for QThread is wrong :) and
@@ -143,6 +143,8 @@
     connect(&gcode, SIGNAL(setVisualLivenessCurrPos(bool)), ui->wgtVisualizer, SLOT(setVisualLivenessCurrPos(bool)));
     connect(&gcode, SIGNAL(setVisCurrLine(int)), ui->wgtVisualizer, SLOT(setVisCurrLine(int)));
     connect(&gcode, SIGNAL(setLcdState(bool)), this, SLOT(setLcdState(bool)));
+/// LETARTARE
+	connect(&gcode, SIGNAL(setVersionGrbl(QString)), ui->GrblVersion, SLOT(setText(QString)));
 
     connect(&runtimeTimer, SIGNAL(setRuntime(QString)), ui->outputRuntime, SLOT(setText(QString)));
 
@@ -296,7 +298,7 @@
 
     writeSettings();
 
-    info("%s has stopped", GRBL_CONTROLLER_NAME_AND_VERSION);
+    info(qPrintable(tr("%s has stopped")), GRBL_CONTROLLER_NAME_AND_VERSION);
 
     SLEEP(300);
 
@@ -481,6 +483,8 @@
         //ui->comboBoxBaudRate->setEnabled(false);
         //ui->btnOpenPort->setEnabled(false);
         ui->btnGRBL->setEnabled(false);
+// LETARTARE
+		ui->GrblVersion->setText(tr("none"));
 
         // Send event to close the port
         emit closePort(reopen);
@@ -1471,9 +1475,9 @@
     {
         ui->resultingZJogSliderPosition->setText(to);
         if(controlParams.useMm)
-            info("Usr chg: pos=%d new=%d\n", pos, newPos);
+            info(qPrintable(tr("Usr chg: pos=%d new=%d\n")), pos, newPos);
         else
-            info("Usr chg: pos=%.1f new=%.1f\n", (double)pos/10, newPos);
+            info(qPrintable(tr("Usr chg: pos=%.1f new=%.1f\n")), (double)pos/10, newPos);
     }
     else
     {
@@ -1480,9 +1484,9 @@
         ui->verticalSliderZJog->setSliderPosition(CENTER_POS);
         ui->currentZJogSliderDelta->setText("0");
         if(controlParams.useMm)
-            info("Usr chg no slider: %d\n", pos);
+            info(qPrintable(tr("Usr chg no slider: %d\n")), pos);
         else
-            info("Usr chg no slider: %.1f\n", (double) pos/10);
+            info(qPrintable(tr("Usr chg no slider: %.1f\n")), (double) pos/10);
     }
 }
 
@@ -1491,18 +1495,18 @@
     sliderPressed = true;
     if (workCoordinates.stoppedZ && workCoordinates.sliderZIndex == sliderZCount)
     {
-        info("Pressed and stopped\n");
+        info(qPrintable(tr("Pressed and stopped\n")));
         sliderTo = workCoordinates.z;
     }
     else
     {
-        info("Pressed not stopped\n");
+        info(qPrintable(tr("Pressed not stopped\n")));
     }
 }
 
 void MainWindow::zJogSliderReleased()
 {
-    info("Released\n");
+    info(qPrintable(tr("Released\n")));
     if (sliderPressed)
     {
         sliderPressed = false;
Index: mainwindow.ui
===================================================================
--- mainwindow.ui	(révision 114)
+++ mainwindow.ui	(copie de travail)
@@ -1758,6 +1758,29 @@
           </property>
          </spacer>
         </item>
+      <item>
+         <widget class="QLabel" name="labelGrblVersion">
+          <property name="text">
+           <string>Firmware :</string>
+          </property>
+         </widget>
+        </item>
+        <item>
+         <widget class="QLabel" name="GrblVersion">
+          <property name="font">
+           <font>
+            <weight>75</weight>
+            <bold>true</bold>
+           </font>
+          </property>
+          <property name="midLineWidth">
+           <number>-1</number>
+          </property>
+          <property name="text">
+           <string>none</string>
+          </property>
+         </widget>
+        </item>
        </layout>
       </item>
      </layout>
@@ -1770,7 +1793,7 @@
      <x>0</x>
      <y>0</y>
      <width>972</width>
-     <height>21</height>
+  <height>23</height>
     </rect>
    </property>
    <widget class="QMenu" name="menuHelp">
Index: README
===================================================================
--- README	(révision 114)
+++ README	(copie de travail)
@@ -20,6 +20,16 @@
 Details on how to do this can be found here:
 http://zapmaker.org/
 
+V3.6.1->8c1   (LETARTARE proposals)
+
+- Improved to the same card 3 and 4 axis detection
+- Displays the version of Grbl
+- Displays line numbers if the uploaded file does not have
+- Updating of the French translation
+- Works with versions Grbl :
+   1- Three axes 0.8c, 0.9D,
+   2- Four axes 0845, future 0.8c1 (0.8c1 axes)
+
 V3.6.1
 Executable release rollup
 
